ocs_ci/resiliency/network_faults.py

```diff
@@ -0,0 +1,246 @@
+            f"Fault duration: {self.duration}s"
+        )
+
+    def _get_node_network_interface(self, node_name, interface_type="default"):
+        """
+        Returns the specified network interface name on the given node.
+
+        Args:
+            node_name (str): The name of the node.
+            interface_type (str): Type of interface to fetch ("default" or "ovn").
+
+        Returns:
+            str: Name of the network interface.
+        """
+        if interface_type == "default":
+            cmd = "ip route | awk '/^default/ {print $5}'"
+            log.debug(f"Getting default network interface on node {node_name}")
+        elif interface_type == "ovn":
+            cmd = "ip -o link show | grep ovn | awk -F: '{print $2}'"
+            log.debug(f"Getting OVN network interface on node {node_name}")
+        else:
+            raise ValueError(f"Unsupported interface type: {interface_type}")
+
+        output = self.ocp_obj.exec_oc_debug_cmd(node=node_name, cmd_list=[cmd])
+        return output.strip()
+
+    def _apply_fault(self, description, netem_command):
+        """
+        Helper to apply a network fault to all nodes.
+
+        Args:
+            description (str): Human-readable fault description.
+            netem_command (str): tc netem command string.
+        """
+
+        list_nodes = (
+            random.sample(self.nodes, random.choice([1, 2]))
+            if self.interface == "ovn"
+            else self.nodes
+        )
+
+        for node in list_nodes:
+            log.info(
+                f"NODE: {node.name} Injecting {description} on interface '{self.interface}'"
+            )
+            cmd = f"tc qdisc add dev {self.interface} root netem {netem_command}"
+            try:
+                self.ocp_obj.exec_oc_debug_cmd(node=node.name, cmd_list=[cmd])
+            except CommandFailed:
+                log.error(
+                    f"Failed to apply network fault '{description}' on node {node.name}"
+                )
+                continue
+        log.info(f"{description} active for {self.duration}s")
+        time.sleep(self.duration)
+        self._remove_faults_all_nodes()
+
+    def _remove_faults_all_nodes(self):
+        """
+        Removes all faults from all nodes and waits 5 seconds after cleanup.
+        """
+        for node in self.nodes:
+            log.info(
+                f"Removing all network faults from node {node.name} interface '{self.interface}'"
+            )
+            cmd = f"tc qdisc del dev {self.interface} root"
+            try:
+                self.ocp_obj.exec_oc_debug_cmd(node=node.name, cmd_list=[cmd])
+            except CommandFailed:
+                log.error(
+                    f"Failed to remove network faults from node {node.name} interface '{self.interface}'"
+                )
+                continue
+        log.info("Waiting for 5 seconds to ensure all faults are removed")
```

Comment:
Should we maybe check that faults are removed? 
`tc qdisc show`

Reply:
Added suggested change

Reply:
Done